nomos-services/storage/src/api/backend/rocksdb/da.rs

```diff
@@ -199,6 +200,47 @@ impl StorageDaApi for RocksBackend {
         )
     }
 
+    async fn store_providerid_mappings(
+        &mut self,
+        mappings: HashMap<Self::Id, ProviderId>,
+    ) -> Result<(), Self::Error> {
+        let mut key_provider_map = HashMap::new();
+
+        for (peer_id, provider_id) in mappings {
+            let provider_key = key_bytes(DA_PROVIDER_MAPPINGS_PREFIX, peer_id.to_bytes());
+            let serialized_provider_id =
+                <()>::serialize(&provider_id).expect("Serialization of ProviderId should not fail");
+            key_provider_map.insert(provider_key, serialized_provider_id);
+        }
+
+        self.bulk_store(key_provider_map).await.map_err(|e| {
```

Comment:
Probably we should make a version of bulkstore that doesnt own a type per se. But change it to take an `IntoIterator` instead. So we do not need to feed an inmemory hashmap for this kind of stuff.
cc @andrussal

Reply:
https://github.com/logos-co/nomos/pull/1781/commits/908f3f359c81e3e52c4e2a07cf101637863d2cc8